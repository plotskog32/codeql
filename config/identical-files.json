{
    "C++ IR Instruction": [
        "cpp/ql/src/semmle/code/cpp/ir/internal/raw_ir/Instruction.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/ssa_ir/Instruction.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/aliased_ssa_ir/Instruction.qll"
    ],
    "C++ IR IRBlock": [
        "cpp/ql/src/semmle/code/cpp/ir/internal/raw_ir/IRBlock.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/ssa_ir/IRBlock.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/aliased_ssa_ir/IRBlock.qll"
    ],
    "C++ IR IRVariable": [
        "cpp/ql/src/semmle/code/cpp/ir/internal/raw_ir/IRVariable.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/ssa_ir/IRVariable.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/aliased_ssa_ir/IRVariable.qll"
    ],
    "C++ IR FunctionIR": [
        "cpp/ql/src/semmle/code/cpp/ir/internal/raw_ir/FunctionIR.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/ssa_ir/FunctionIR.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/aliased_ssa_ir/FunctionIR.qll"
    ],
    "C++ IR OperandTag": [
        "cpp/ql/src/semmle/code/cpp/ir/internal/raw_ir/OperandTag.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/ssa_ir/OperandTag.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/aliased_ssa_ir/OperandTag.qll"
    ],
    "C++ IR IRImpl": [
        "cpp/ql/src/semmle/code/cpp/ir/internal/raw_ir/IRImpl.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/ssa_ir/IRImpl.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/aliased_ssa_ir/IRImpl.qll"
    ],
    "C++ IR IRSanityImpl": [
        "cpp/ql/src/semmle/code/cpp/ir/internal/raw_ir/IRSanityImpl.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/ssa_ir/IRSanityImpl.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/aliased_ssa_ir/IRSanityImpl.qll"
    ],
    "C++ IR PrintIRImpl": [
        "cpp/ql/src/semmle/code/cpp/ir/internal/raw_ir/PrintIRImpl.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/ssa_ir/PrintIRImpl.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/aliased_ssa_ir/PrintIRImpl.qll"
    ],
    "C++ SSA AliasAnalysis": [
        "cpp/ql/src/semmle/code/cpp/ir/internal/ssa_ir/AliasAnalysis.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/aliased_ssa_ir/AliasAnalysis.qll"
    ],
    "C++ SSA SimpleSSA": [
        "cpp/ql/src/semmle/code/cpp/ir/internal/ssa_ir/SimpleSSA.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/aliased_ssa_ir/SimpleSSA.qll"
    ],
    "C++ SSA IRBlockConstruction": [
        "cpp/ql/src/semmle/code/cpp/ir/internal/ssa_ir/IRBlockConstruction.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/aliased_ssa_ir/IRBlockConstruction.qll"
    ],
    "C++ SSA SSAConstruction": [
        "cpp/ql/src/semmle/code/cpp/ir/internal/ssa_ir/SSAConstruction.qll",
        "cpp/ql/src/semmle/code/cpp/ir/internal/aliased_ssa_ir/SSAConstruction.qll"
    ]
}
