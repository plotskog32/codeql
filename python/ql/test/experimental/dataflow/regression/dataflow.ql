import config

from
  DataFlow::Node source,
  DataFlow::Node sink
where
  // source != sink and 
  exists(TestConfiguration cfg | cfg.hasFlow(source, sink))
select
  source, sink
