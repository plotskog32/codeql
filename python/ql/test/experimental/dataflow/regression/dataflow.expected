| test.py:3:10:3:15 | ControlFlowNode for SOURCE | test.py:3:10:3:15 | ControlFlowNode for SOURCE |
| test.py:6:9:6:14 | ControlFlowNode for SOURCE | test.py:7:10:7:10 | ControlFlowNode for s |
| test.py:10:12:10:17 | ControlFlowNode for SOURCE | test.py:13:10:13:12 | ControlFlowNode for arg |
| test.py:10:12:10:17 | ControlFlowNode for SOURCE | test.py:17:10:17:10 | ControlFlowNode for t |
| test.py:20:9:20:14 | ControlFlowNode for SOURCE | test.py:13:10:13:12 | ControlFlowNode for arg |
| test.py:37:13:37:18 | ControlFlowNode for SOURCE | test.py:41:14:41:14 | ControlFlowNode for t |
| test.py:62:13:62:18 | ControlFlowNode for SOURCE | test.py:13:10:13:12 | ControlFlowNode for arg |
| test.py:67:13:67:18 | ControlFlowNode for SOURCE | test.py:13:10:13:12 | ControlFlowNode for arg |
| test.py:76:9:76:14 | ControlFlowNode for SOURCE | test.py:78:10:78:10 | ControlFlowNode for t |
| test.py:128:13:128:18 | ControlFlowNode for SOURCE | test.py:132:14:132:14 | ControlFlowNode for t |
| test.py:159:10:159:15 | ControlFlowNode for SOURCE | test.py:160:14:160:14 | ControlFlowNode for t |
| test.py:163:9:163:14 | ControlFlowNode for SOURCE | test.py:165:10:165:10 | ControlFlowNode for s |
| test.py:178:9:178:14 | ControlFlowNode for SOURCE | test.py:180:14:180:14 | ControlFlowNode for t |
| test.py:178:9:178:14 | ControlFlowNode for SOURCE | test.py:182:14:182:14 | ControlFlowNode for t |
| test.py:178:9:178:14 | ControlFlowNode for SOURCE | test.py:184:14:184:14 | ControlFlowNode for t |
| test.py:178:9:178:14 | ControlFlowNode for SOURCE | test.py:186:14:186:14 | ControlFlowNode for t |
| test.py:195:9:195:14 | ControlFlowNode for SOURCE | test.py:197:14:197:14 | ControlFlowNode for t |
| test.py:195:9:195:14 | ControlFlowNode for SOURCE | test.py:199:14:199:14 | ControlFlowNode for t |
| test.py:217:15:217:20 | ControlFlowNode for SOURCE | test.py:219:14:219:20 | ControlFlowNode for tainted |
| test.py:217:15:217:20 | ControlFlowNode for SOURCE | test.py:220:10:220:16 | ControlFlowNode for tainted |
| test.py:223:15:223:20 | ControlFlowNode for SOURCE | test.py:226:10:226:16 | ControlFlowNode for tainted |
| test.py:229:15:229:20 | ControlFlowNode for SOURCE | test.py:231:14:231:20 | ControlFlowNode for tainted |
| test.py:229:15:229:20 | ControlFlowNode for SOURCE | test.py:233:14:233:20 | ControlFlowNode for tainted |
